src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -1597,7 +1610,7 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     const typedResult: CanTransferByPartitionData = {
       statusCode: result[0],
       reasonCode: bytes32ToString(result[1]),
-      partition: bytes32ToString(result[2]),
+      partition: bytes32ToString(result[2]) === Partition.Unlocked ? Partition.Unlocked : Partition.Undefined,
```

Comment:
This check should be refactored into some kind of utility function (to be easily extensible later when more partitions are added). The function can be called `parsePartitionValue` or something along those lines, and should

- Receive the string value
- Use a switch statement and check for each possible partition value
- Special case for '0': return `Partition.Undefined`
- Default case: throw an error. This case means the core team added a new partition and our wrappers haven't been adapted